airflow/api_fastapi/app.py

```diff
@@ -28,6 +30,18 @@
 app: FastAPI | None = None
 
 
+@asynccontextmanager
+async def lifespan(app: FastAPI):
+    async with AsyncExitStack() as stack:
+        for route in app.routes:
+            if isinstance(route, Mount) and isinstance(route.app, FastAPI):
+                await stack.enter_async_context(
+                    route.app.router.lifespan_context(route.app),
```

Comment:
Does this need an `if route.app.lifespan`?

Reply:
Possibly. I don't think so though as I didn't add one to core and the unit test didn't barf. But it might be a good addition anyway.

Reply:
Core isn't a separate app I don't think -- it's some routers

Reply:
You're right. Doing a quick sanity check

Reply:
It will always have a lifespan https://github.com/encode/starlette/blob/c2e3a39b09a613553ee03586589ed9cd0fbf07f3/starlette/routing.py#L612 but I can add a check in if you want to ensure there's no regression.

Reply:
I think it should be safe to merge as is.